src/Label.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import cx from 'classnames';
+
+/**
+ * @module Label
+ */
+class Label extends React.Component {
+	render() {
+		const {
+			inputId,
+			text,
+			isRequired,
+			children,
+			className,
+			...other
+		} = this.props;
+
+		const classNames = cx(
+			{ required : isRequired },
+			className
+		);
+
+		return (
+			<label className={classNames}
+				htmlFor={inputId}
+				{...other}>
+				{text}
+				{children}
+			</label>
+		);
+	}
+}
+
+Label.propTypes = {
+};
+
+export default Label;
```

Comment:
I like the compound component `TextField` setup, although at the moment it doesn't look like the `<input>` itself needs much special treatment, so I'd probably avoid making a separate component for that.